pam/return.go

```diff
@@ -1,54 +1,41 @@
 package main
 
+import (
+	"github.com/msteinert/pam"
+)
+
 // Various signalling return messaging to PAM.
 
-// pamSuccess signals PAM module to return PAM_SUCCESS and Quit tea.Model.
+type pamReturnStatus interface {
+	Status() pam.StatusError
+	Error() string
+}
+
+// TODO: Remove this when we've tests coverage that ensures this is the case.
+// ensure that [pam.NewTransactionError] implements [pamReturnStatus].
+var _ pamReturnStatus = pam.NewTransactionError(pam.ErrAbort, nil)
+
+// pamStatus signals PAM module to return with provided pam.Success and Quit tea.Model.
 type pamSuccess struct {
+	pam.TransactionError
 	brokerID string
 }
 
-// String returns the string of pamSuccess.
-func (err pamSuccess) String() string {
-	return ""
+// newPamSuccess returns a new pamSuccess.
+func newPamSuccess(brokerID string) pamReturnStatus {
+	return pamSuccess{nil, brokerID}
 }
 
-// pamIgnore signals PAM module to return PAM_IGNORE and Quit tea.Model.
+// pamIgnore signals PAM module to return pam.Ignore and Quit tea.Model.
 type pamIgnore struct {
+	pam.TransactionError
 	localBrokerID string // Only set for local broker to store it globally.
-	msg           string
-}
-
-// String returns the string of pamIgnore message.
-func (err pamIgnore) String() string {
-	return err.msg
-}
-
-// pamAbort signals PAM module to return PAM_ABORT and Quit tea.Model.
-type pamAbort struct {
-	msg string
-}
-
-// String returns the string of pamAbort message.
-func (err pamAbort) String() string {
-	return err.msg
-}
-
-// pamSystemError signals PAM module to return PAM_SYSTEM_ERROR and Quit tea.Model.
-type pamSystemError struct {
-	msg string
-}
-
-// String returns the string of pamSystemError message.
-func (err pamSystemError) String() string {
-	return err.msg
-}
-
-// pamAuthError signals PAM module to return PAM_AUTH_ERROR and Quit tea.Model.
-type pamAuthError struct {
-	msg string
 }
 
-// String returns the string of pamAuthError message.
-func (err pamAuthError) String() string {
-	return err.msg
+// newPamIgnore returns a new pamIgnore.
+func newPamIgnore(localBrokerID string, err error) pamReturnStatus {
+	return pamIgnore{
```

Comment:
Use field tags to make the struct more future proof and don’t rely on the order.